draft-regext-rdap-extensions.md

```diff
@@ -386,125 +484,187 @@ that example could be written as:
+new types in the RDAP JSON Values Registry (see
+(#rdap_json_values_registry)).
 
-# Profile and Marker Extensions {#profiles_and_markers}
+# Extension Implementer Considerations {#extension_implementer_considerations}
 
-Extensions are not required to extend the JSON or URL components of RDAP.
+## Redirects {#redirects_implementer}
 
-While the RDAP extension mechanism was created to extend RDAP queries
-and/or responses, extensions can also be used to signal server policy 
-(for example, specifying the conditions of use for existing
-response structures). Extensions that are primarily about signaling
-server policy are often called "profiles".
-
-Some extensions exist to denote the usage of values placed into an
-IANA registry, such as the IANA RDAP registries, or the usage of extensions
-for specifications used in RDAP responses such as extended vCard/jCard properties.
-Such extensions exist to "mark" these usages and are often called "marker"
-extensions.
+[@!RFC7480] describes the use of redirects in RDAP. Redirects are prominent
+in the discovery of authoritative RIR servers, as the process outlined in
+[@!RFC9224], which uses IANA allocations, does not account for transfers of
+resources between RIRs. [@!RFC7480, Section 4.3] instructs servers to ignore
+unknown query parameters. As it relates to issuing URLs for redirects, servers
+MUST NOT blindly copy query parameters from a request to a redirect URL as
+query parameters may contain sensitive information, such as security credentials,
+not relevant to the target server of the URL. Following the advice in [@!RFC7480],
+servers SHOULD only place query parameters in redirect URLs when it is known
+by the origin server (the server issuing the redirect) that the target server
+(the server referenced by the redirect) can process the query parameter and the
+contents of the query parameter are appropriate to be received by the target.
 
-For example, an extension may be used to signal desired processing of
-a `rel` attribute in a "links" array, where the `rel` value is registered in
-the Link Relations Registry (<https://www.iana.org/assignments/link-relations/link-relations.xhtml>).
+# Extension Author Considerations {#extension_author_considerations}
+
+## Redirects {#redirects_author}
+
+As it is unlikely that every server in a cross-authority, redirect
+scenario will be upgraded to process every new extension, extensions
+should not rely on query parameters alone to convey information about
+a resource, as query parameters are not guaranteed to survive a
+redirect.
+
+This does not mean extensions are prohibited from using query
+parameters, but rather that the use of query parameters must be
+applied for the scenarios appropriate for the use of the extension.
+Therefore, extensions SHOULD NOT rely on query parameters when the
+extension is to be used in scenarios requiring clients to find
+authoritative servers, such as that described above, or other
```

Comment:
Would be it better to reference the section when saying "such as that described above"?

Reply:
I think this text is referring either to the previous paragraph ("As it is unlikely...") or the one before that from the previous version ("RFC7480 describes the use...").  But in any event, simply removing "such as that described above" works as well, so this has been updated to remove that part.